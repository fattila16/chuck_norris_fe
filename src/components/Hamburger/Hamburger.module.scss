@import 'src/assets/scss/color-palette';

.hamburger{
    cursor: pointer;
    span{
        width:30px;
        height:4px;
        background: map-get($active-theme, fawn);
        display:block;
        transition: all 0.3s ease;
        &:first-child {
            transform: translateY(-5px);
        }
        &:nth-child(2) {
            opacity: 1;
        }
        &:last-child {
            transform: translateY(5px);
        }
    }
    &.open {
        span {
            &:first-child {
                transform: translateY(4px) rotate(45deg);
            }
            &:nth-child(2) {
                transform: translateX(30px);
                opacity: 0;
            }
            &:last-child {
                transform: translateY(-4px) rotate(-45deg);
            }
        }
    }
}
