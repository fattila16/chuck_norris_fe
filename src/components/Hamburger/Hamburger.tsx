import React from 'react';
import css from './Hamburger.module.scss';

interface HamburgerProps {
    open: boolean;
    setOpen: (open: boolean) => void;
}

const Hamburger: React.FC<HamburgerProps> = (props: HamburgerProps) => {
    const { open, setOpen } = props;
    return (
        <div
            onClick={(): void => setOpen(!open)}
            onKeyPress={(): void => setOpen(!open)}
            className={`${css.hamburger} ${open ? css.open : ''}`}
            role="menu"
            tabIndex={0}
        >
            <span />
            <span />
            <span />
        </div>
    );
};

export default Hamburger;
