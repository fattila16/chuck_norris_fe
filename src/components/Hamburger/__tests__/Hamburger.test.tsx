import React from 'react';
import { shallow } from 'enzyme';
import Hamburger from '../Hamburger';

describe('[Hamburger]', () => {
    it('Should render Hamburger with classNames: hamburger open, if open is true', () => {
        const component = shallow(<Hamburger open={true} setOpen={jest.fn()} />);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('Should render Hamburger with classNames: hamburger, if open is false', () => {
        const component = shallow(<Hamburger open={false} setOpen={jest.fn()} />);
        expect(component.getElements()).toMatchSnapshot();
    });
});
