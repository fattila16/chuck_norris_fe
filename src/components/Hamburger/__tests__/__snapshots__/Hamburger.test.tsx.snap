// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Hamburger] Should render Hamburger with classNames: hamburger open, if open is true 1`] = `
Array [
  <div
    className="hamburger open"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="menu"
    tabIndex={0}
  >
    <span />
    <span />
    <span />
  </div>,
]
`;

exports[`[Hamburger] Should render Hamburger with classNames: hamburger, if open is false 1`] = `
Array [
  <div
    className="hamburger "
    onClick={[Function]}
    onKeyPress={[Function]}
    role="menu"
    tabIndex={0}
  >
    <span />
    <span />
    <span />
  </div>,
]
`;
